File: back/src/controller/login.controller.js
const bcrypt = require('bcrypt');
const db = require('../db/conection');

const postLogin = async (req, res) => {
  const { email, contraseña } = req.body;
  if (!email || !contraseña) {
    return res.status(400).json({ error: 'Faltan datos obligatorios' });
  }

  try{
    let sql = 'SELECT * FROM usuarios WHERE email = ?';
    const [resultados] = await db.query(sql, [email]);
    
    if (resultados.length === 0) {
      return res.status(401).json({ error: 'El usuario no existe' });
    }
    const usuario = resultados[0]
    console.log(`ID de usuario: ${usuario.id} linea 18 login.controller.js`); // Debugging line to check user ID
    const contraseñaValidada = await bcrypt.compare(contraseña, usuario.contraseña);
    if (!contraseñaValidada) {
      return res.status(401).json({ error: 'Contraseña incorrecta' });
    }
    console.log('Inicio de sesión exitoso:', usuario);
    req.session.usuarioId = usuario.id;
    req.session.nombre = usuario.nombre;
    req.session.apellidos = usuario.apellidos;
    req.session.email = usuario.email;
    req.session.save((err) => {
      if (err) {
        console.error('Error al guardar la sesión:', err);
        return res.status(500).json({ error: 'Error interno del servidor' });
      }
    });
    return res.status(200).json({
      usuario: {
        id: usuario.id,
        nombre: usuario.nombre,
        apellidos: usuario.apellidos,
        email: usuario.email
      },
      mensaje: 'Inicio de sesión exitoso'
    });
  } catch (error) {
    console.error(error);
    return res.status(500).json({ error: 'Error interno del servidor' });
  }
}

module.exports = { postLogin };

File: ListaDeLaCompra_copia/back_py/controller/login_controller.py
import bcrypt
from db.conexion import get_connection
from flask import session

def post_login(email, contrasena):
    if not email or not contrasena:
        return {"error": "Faltan datos obligatorios"}, 400
    conn = get_connection()
    cursor = conn.cursor(dictionary=True)
    try:
        sql = "SELECT * FROM usuarios WHERE email = %s"
        cursor.execute(sql, (email,))
        resultados = cursor.fetchall()
        if len(resultados) == 0:
            return {"error": "El usuario no existe"}, 401
        usuario = resultados[0]
        if not bcrypt.checkpw(contrasena.encode(), usuario["contrasena"].encode()):
            return {"error": "Contrasena incorrecta"}, 401
        session["usuario_id"] = usuario["id"]
        session["nombre"] = usuario["nombre"]
        session["apellidos"] = usuario["apellidos"]
        session["email"] = usuario["email"]
        return {
            "usuario": {
                "id": usuario["id"],
                "nombre": usuario["nombre"],
                "apellidos": usuario["apellidos"],
                "email": usuario["email"]
            },
            "mensaje": "Inicio de sesion exitoso"
        }, 200
    except Exception:
        return {"error": "Error interno del servidor"}, 500
    finally:
        cursor.close()
        conn.close()
